File: exercism-python/clock/clock.py
class Clock(object):
    """
    A clock implementation that handles times without dates.
    """

    TOTAL_MINUTE_OF_A_DAY = 24 * 60

    def __init__(self, hour, minute):
        self.minutes = (minute + 60 * hour) % self.TOTAL_MINUTE_OF_A_DAY

    def __repr__(self):
        return '{:02d}:{:02d}'.format(self.minutes // 60, self.minutes % 60)

    def __eq__(self, other):
        """
        Two clocks that represent the same time should be equal to each other.
        """
        return self.minutes == other.minutes

    def __add__(self, minutes):
        """
        Add minutes to the current time.
        """
        self.minutes = (self.minutes + minutes) % self.TOTAL_MINUTE_OF_A_DAY
        return self

    def __sub__(self, minutes):
        """
        Subtract minutes to the current time.
        """

        self.minutes = (self.minutes - minutes) % self.TOTAL_MINUTE_OF_A_DAY
        return self

File: exercism-go/clock/clock.go
// Package clock implements a clock that handles times without dates.
package clock

import "fmt"

// Clock defines the clock type with hour and minute.
type Clock struct {
	hour, minute int
}

const minOfDay = 24 * 60

// New returns a new clock.
func New(h, m int) Clock {
	totalMin := (minOfDay + (h*60+m)%minOfDay) % minOfDay
	h = totalMin / 60
	m = totalMin % 60

	return Clock{h, m}
}

// Add returns a new clock after adding minutes to it.
func (c Clock) Add(m int) Clock {
	return New(c.hour, c.minute+m)
}

// Subtract returns a new clock after subtracting minutes to it.
func (c Clock) Subtract(m int) Clock {
	return New(c.hour, c.minute-m)
}

// String returns a string that represents the clock in format of [hh]:[mm].
func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hour, c.minute)
}
